src/lib.rs

```diff
@@ -405,16 +426,16 @@ impl PctString {
     /// println!("{}", pct_string.as_str()); // => Hello World%21
     /// ```
     pub fn encode<I: Iterator<Item = char>, E: Encoder>(src: I, encoder: E) -> PctString {
+        use std::fmt::Write;
+
+        let mut buf = String::with_capacity(4);
```

Comment:
Again we can avoid an unnecessary allocation here since we know the actual size of the buffer. Instead you can use a `[u8; 4]` buffer combined with the [`encode_utf8`](https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8) function of the standard library.